src/main/scala/no/ndla/listingapi/controller/NdlaController.scala

```diff
@@ -52,6 +55,9 @@ abstract class NdlaController extends ScalatraServlet with NativeJsonSupport wit
     case o: OptimisticLockException => Conflict(body=Error(Error.RESOURCE_OUTDATED, o.getMessage))
     case e: CoverAlreadyExistsException => Conflict(body=Error(Error.ALREADY_EXISTS, e.getMessage, Some(e.id)))
     case rw: ResultWindowTooLargeException => UnprocessableEntity(body = Error(Error.WINDOW_TOO_LARGE, rw.getMessage))
+    case _: PSQLException =>
+      ConnectionPool.singleton(new DataSourceConnectionPool(ComponentRegistry.dataSource))
```

Comment:
Synes det blir litt uryddig at du nå initialiserer connection pool to steder (både her, og i `ComponentRegistry`). Kunne vi samlet det? F.eks. med en funksjon i `ComponentRegistry` som gjør connection pool initialiseringen både ved oppstart og ved feilmelding...